apm/src/app/user/user.component.ts

```diff
@@ -0,0 +1,51 @@
+import { Component, OnInit } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { Router } from '@angular/router';
+import { UserService } from './user.service';
+
+@Component({
+  templateUrl: './user.component.html',
+  styleUrls: ['./user.component.css']
+})
+export class UserComponent implements OnInit {
+
+  public editForm!: FormGroup;
```

Comment:
try to avoid using `!` anywhere. The value for this technically speaking (without considering angular internal working) is undefined unless the ngOnInit method is called explicitly. So although we are saying the the editForm is of type `FormGroup` it actually is `FormGroup | undefined`

Reply:
Having all that said, please do not take all that what i say as something to simply do, Someplaces it is also a development convention that people have considering the internal working of Angular